### DISCUSSION TITLE
Reason get_artist sometimes doesn't return songs

### DISCUSSION THREAD

**Discussion by Schtenk** [sigma67/ytmusicapi#478] [General]
Hey, i'm not sure this is the correct place to put this but I think I know why get_artist sometimes doesn't return songs.

There seems to be two different "channelIds" per artist.
both of these works to get an artist except one does not include songs.

- ### ChannelId

This i is the one this api saves as "channelId" 

This type of Id for for Mariah Carey is: UCurpiDXSkcUbgdMwHNZkrCg

This is the one found at the jsonpath:
header.musicImmersiveHeaderRenderer.subscriptionButton.subscribeButtonRenderer.channelId

Using this one in a new request no songs are included in the response.

- ### ChannelId/ChannelBrowseId

This is the one that youtube itself seems to mostly use which can only be found on songs/albums/singles or "More" buttons in the response.

This type of Id for Mariah Carey is: UCKRnq8aBOCanYlffje7HyvA

On songs/album/singles it is found in the above form^

From your library "More" button its in the form of:
- MPLAUCKRnq8aBOCanYlffje7HyvA

And on all other "More" Buttons its in the form of:
- MPADUCKRnq8aBOCanYlffje7HyvA

The full jsonpath may look something like this (MPAD version):
contents.singleColumnBrowseResultsRenderer.tabs[0].tabRenderer.
content.sectionListRenderer.contents[1].musicCarouselShelfRenderer.
header.musicCarouselShelfBasicHeaderRenderer.moreContentButton.
buttonRenderer.navigationEndpoint.browseEndpoint.browseId

Using this one in a new request songs are included in the response.

**Comment by Schtenk** [comment]
Responses are also slightly different, for example ShuffleId is under "watchPlaylistEndpoint" (for channelId) instead of "watchEndpoint" (for channelBrowseId), same for RadioId.

**Comment by Schtenk** [comment]
found that videos are also missing when using "channelId"

**Comment by sigma67** [comment]
See for yourself:

https://music.youtube.com/channel/UCurpiDXSkcUbgdMwHNZkrCg

https://music.youtube.com/channel/UCKRnq8aBOCanYlffje7HyvA
